File: src/main/java/com/baulsupp/oksocial/util/FileContent.java
package com.baulsupp.oksocial.util;

import java.io.File;
import java.io.IOException;
import java.nio.charset.StandardCharsets;
import org.apache.commons.io.FileUtils;
import org.apache.commons.io.IOUtils;

public class FileContent {

  public static byte[] readParamBytes(String param) throws IOException {
    if (param.equals("@-")) {
      return IOUtils.toByteArray(System.in);
    }
    if (param.startsWith("@")) {
      return FileUtils.readFileToByteArray(new File(param.substring(1)));
    } else {
      return param.getBytes(StandardCharsets.UTF_8);
    }
  }

  public static String readParamString(String param) throws IOException {
    if (param.equals("@-")) {
      return IOUtils.toString(System.in);
    }
    if (param.startsWith("@")) {
      return FileUtils.readFileToString(new File(param.substring(1)));
    } else {
      return param;
    }
  }
}

File: src/main/kotlin/com/baulsupp/oksocial/util/FileContent.kt
package com.baulsupp.oksocial.util

import com.baulsupp.oksocial.kotlin.IO
import kotlinx.coroutines.experimental.async
import org.apache.commons.io.IOUtils
import java.io.File
import java.nio.charset.Charset
import java.nio.charset.StandardCharsets

object FileContent {

  suspend fun readParamBytes(param: String): ByteArray {
    if (param == "@-") {
      return IOUtils.toByteArray(System.`in`)
    }
    return if (param.startsWith("@")) {
      async(IO) { File(param.substring(1)).readBytes() }.await()
    } else {
      param.toByteArray(StandardCharsets.UTF_8)
    }
  }

  suspend fun readParamString(param: String): String {
    if (param == "@-") {
      return IOUtils.toString(System.`in`, Charset.defaultCharset())
    }
    return if (param.startsWith("@")) {
      async(IO) { File(param.substring(1)).readText() }.await()
    } else {
      param
    }
  }

}
